### PR TITLE
add SIG VMware charter.md

### PR DESCRIPTION
**Pull request by cantbewong**
### TODO create new original SIG charter

- [x] communicate within VMware SIG and get feedback
- [x] solicit steering committee review

### CODE REVIEW

**Comment by quinton-hoole** [comment]
(for steering committee) This charter seems sane, but presumably this effort needs to be a sub-project of [sig-cloud-provider](https://github.com/kubernetes/community/tree/master/sig-cloud-provider), and there's already a cloud-provider-vsphere subproject listed there.  Could you please clarify the intended relationship between the cloud-provider-vsphere subproject of sig-cloud-provider, and this sig-vmware?  It seems that the two should be collapsed into one, unless there are very compelling reasons why they should be kept separate, in which case those compelling reasons should be explicitly documented here.

**Comment by frapposelli** [comment]
@quinton-hoole the original charter was written months ago, before sig-cloud-provider existed, since then, cloud-provider-vsphere is transitioning under sig-cloud-provider and SIG-VMware charter has been expanded, @cantbewong amended the PR with the changes.

**Comment by philips** [comment]
Thank you for your patience as we work to your SIG charter merged.

Through the process of reviewing the [first 11 charters][charters] the Steering Committee (SC) found a lot of copy/paste language caused by our poorly designed initial charter template. To fix this there is a new [SIG charter template][template] which focuses on the main things we want to see in charters: scope and responsibilities.

Please consider updating to this charter template and putting focus into the scope and responsibilities. SC members will focus more on a deep review of scope more than anything else and using this template will help with that focus.

[charters]: https://docs.google.com/spreadsheets/d/1i_8oM1KlmXo90zIIgNTdOg-M6xesTggFLnTkgmscMlI/edit#gid=0
[template]: https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-charter-template.md

**Comment by frapposelli** [comment]
@cantbewong can you please amend the charter with the template mentioned above?

File: sig-vmware/charter.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The VMware SIG maintains and evolves the ability to run Kubernetes on VMware infrastructure.
+
+## Scope
+
+SIG VMware serves to bring together members of the VMware and Kubernetes community to maintain, support and provide guidance for running Kubernetes on VMware platforms.
+
+One focus area is implementing and maintaining the Kubernetes cloud provider integration code for vSphere.
+
+This SIG will also fill a role beyond implementation of a cloud provider for vSphere. Thus it is appropriate to operate a VMware SIG, rather than expecting that all VMware infrastructure related activity be managed as a working group underneath sig-cloud-provider.
```
Comment (philips) [inline sig-vmware/charter.md]: I don't think this is necessary as the following paragraph outlines a large scope well.

**Comment by philips** [comment]
@cantbewong Are you going to run this by the SIG?

**Comment by cantbewong** [comment]
I am amending the charter to align with the new template. I will make a draft available for review and comment by SIG membership before updating this PR.

File: sig-vmware/charter.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- vRealize Automation Cluster API Provider
+
+#### Cross-cutting and Externally Facing Processes
+
+SIG VMware serves to bring together members of the VMware and Kubernetes community to maintain, support and provide guidance for running Kubernetes on VMware platforms.
+
+The VMware SIG will provide a forum for hosting related architectural planning and discussion. Associated activities related to development, testing, and documentation will be tracked and reported by the SIG.
+
+- The vSphere platform’s availability and resource management can create pools that extend across Kubernetes worker node, management node, and namespace boundaries. Unique features of the vSphere platform related to high availability, load balancing, and resource management (storage, networking, compute) can supplement existing capabilities of Kubernetes, container runtime, and OS functionality.
+
```
Comment (philips) [inline sig-vmware/charter.md]: nit: delete this newline

File: sig-vmware/charter.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+## Roles and Organization Management
+
+This sig follows adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance].
+
+### Subproject Creation
+
+Subprojects associated with this SIG may be created following the procedure described in [sig-governance] as:
+
+> by Federation of Subprojects
```
Comment (philips) [inline sig-vmware/charter.md]: what happened here?

File: sig-vmware/charter.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### In scope
+- Deploying Kubernetes of any type (standard or a packaged distribution from any vendor) on vSphere infrastructure.
+- Utilizing networking and storage features of VMware infrastructure through Kubernetes.
+- Determining and documenting best practices for configuring Kubernetes when running on vSphere infrastructure.
+- Determining and documenting best practices for configuring VMware infrastructure when supporting Kubernetes.
+- Determining and documenting best practices for using common configuration management and orchestration tools for deploying and managing Kubernetes on VMware infrastructure.
+- Discussing bugs and feature requests recorded as Kubernetes or VMware cloud provider issues on GitHub. These issues should be tagged with ``sig/vmware``.
+
+The authoritative source for SIG information is the [sigs.yaml] file.
```
Comment (philips) [inline sig-vmware/charter.md]: can you link here instead? https://github.com/kubernetes/community/tree/master/sig-vmware#subprojects

File: sig-vmware/charter.md (hunk 3 above)
Comment (philips) [reply sig-vmware/charter.md]: sorry for the unclear comment. It is just odd to use a markdown quote here. Can you just do something like:

"Associated subprojects are created following the Federation of Subprojects procedure described in [sig-governance]"

Comment (cantbewong) [reply sig-vmware/charter.md]: got it, will do

**Comment by philips** [comment]
/lgtm
/approve
/hold

Holding for @spiffxp to comment as he wishes. I will release the hold on Friday if he has no comments.

File: sig-vmware/charter.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+The VMware SIG will provide a forum for hosting related architectural planning and discussion. Associated activities related to development, testing, and documentation will be tracked and reported by the SIG.
+
+- The vSphere platform’s availability and resource management can create pools that extend across Kubernetes worker node, management node, and namespace boundaries. Unique features of the vSphere platform related to high availability, load balancing, and resource management (storage, networking, compute) can supplement existing capabilities of Kubernetes, container runtime, and OS functionality.
+  - Users can do some of this via configuration today, but opportunities exist to achieve more efficient, predictable, and dependable service through additional new CRDs, plug-ins, and enhancement proposals (KEPs) related to other Kubernetes components.
+  - Interaction is expected to cut across multiple aspects of Kubernetes in ways not firmly bounded by existing SIG dividing lines (e.g. node, scheduling, scalability).
+
```
Comment (spiffxp) [inline sig-vmware/charter.md]: The first paragraph looks good, as an external process description.  The rest of this section reads to me more like marketing for vSphere than it does a description of how this SIG interacts with other SIGs on cross-cutting efforts.

Seems like this is all justification for why the SIG should continue to exist outside the scope of other SIGs, which IMO could be articulated in scope above, eg:

- "Hosting architectural planning and discussion related to new CRDs, plug-ins and KEPs that allow the vSphere platform to supplement the existing capabilities of Kubernetes, container runtime, and OS functionality"

In here, I would expect to see something more like:
- [We co-own vSphere-specific code related to cluster and machine provisioning with SIG Cluster Lifecycle](https://github.com/kubernetes/community/blob/master/sig-cluster-lifecycle/charter.md#cross-cutting-and-externally-facing-processes)
- We drive changes that cross existing SIG boundaries by collaborating with the appropriate SIGs (eg: node, schedule, scalability, etc.)

**Comment by spiffxp** [comment]
I have one nit I'd like to see addressed, everything else lgtm

File: sig-vmware/charter.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+The VMware SIG is not for discussing bugs or feature requests outside the scope of Kubernetes. For example, the VMware SIG should not be used to discuss or resolve support requests related to VMware products. It should also not be used to discuss topics that other, more specialized SIGs own (to avoid overlap).
+
+## Roles and Organization Management
+
+This sig follows adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance].
+
+### Subproject Creation
+
+Associated subprojects are created following the Federation of Subprojects procedure described in [sig-governance]
```
Comment (spiffxp) [inline sig-vmware/charter.md]: You need to choose which option you're taking.

Comment (cantbewong) [reply sig-vmware/charter.md]: The option shoices are Option 1: by SIG Technical Leads or Option 2: by Federation of Subprojects.. so if the election of the "Federation of Subproject" isn't clear now could you suggest better wording?

Comment (spiffxp) [reply sig-vmware/charter.md]: ack, I misread, I have been used to sigs saying "Option 1" or "Option 2", this is fine

File: sig-vmware/charter.md (hunk 5 above)
Comment (cantbewong) [reply sig-vmware/charter.md]: edited charter in response to your comments, adding more details on interaction with sig-cluster-lifecycle, sig-cloud-provider, and sig-storage and removing "marketing text"

File: sig-vmware/charter.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+- vRealize Automation Cluster API Provider
+
+Code locations and projects are defined in [SIG Subprojects](https://github.com/kubernetes/community/tree/master/sig-vmware#subprojects)
+
+#### Cross-cutting and Externally Facing Processes
+
+SIG VMware serves to bring together members of the VMware and Kubernetes community to maintain, support and provide guidance for running Kubernetes on VMware platforms.
+
+The VMware SIG will provide a forum for hosting related architectural planning and discussion. Associated activities related to development, testing, and documentation will be tracked and reported by the SIG.
+- The vSphere platform’s availability and resource management can create pools that extend across Kubernetes worker node, management node, and namespace boundaries. Unique features of the vSphere platform related to high availability, load balancing, and resource management (storage, networking, compute) can supplement existing capabilities of Kubernetes, container runtime, and OS functionality.
```
[the remark this answers is not in the document]
Comment (spiffxp) [reply sig-vmware/charter.md]: That said, your edit leaves me feeling less strongly about this.  I'll defer to the judgement of another steering committee reviewer.  @philips is out this week, I will raise during tomorrow's steering committee meeting

[the remark this answers is not in the document]
Comment (smarterclayton) [reply sig-vmware/charter.md]: I think this looks ok - GCP or AWS or OpenStack would reasonably write the same thing (I can find-replace for all of those and it reads the same).

**Comment by spiffxp** [comment]
/hold cancel
/lgtm
thanks for the extra look @smarterclayton

**Pull request merged**